www/js/coolhole-util.js

```diff
@@ -19,18 +19,284 @@ function coolholePostFormatMessage(data, last, div, time, name, message) {
+    return new Date() >= messageTime;
+}
+
+//-----------------------------------------------------------
+// [START] SOUND EFFECTS
+//-----------------------------------------------------------
+// Define options object
+const SFX = {
+    global: {
+      checkboxId: 'chatOptions-sfx-global-cb',
+      localStorageTag: 'sfxGlobalEnabled',
+      getState: {},
+      setState: {}
+    },
+    mod: {
+      checkboxId: 'chatOptions-sfx-mod-cb',
+      localStorageTag: 'sfxModEnabled',
+      getState: {},
+      setState: {}
+    },
+    stack: {
+      checkboxId: 'chatOptions-sfx-stack-cb',
+      localStorageTag: 'sfxStackEnabled',
+      getState: {},
+      setState: {}
+    },
+    secretary: {} // empty object since no ui tied to it
+};
+
+  
+// 06-29-2024 Miles - temporarily commented this out until we have coolhole options built.
+// Initialize
+// [ 'global', 'mod', 'stack' ].forEach( type => {
+  
+//     // Bind Element
+//     SFX[ type ].checkbox = document.getElementById( SFX[ type ].checkboxId );
+  
+//     // Get Checkbox State Function
+//     SFX[ type ].getState.cb = () => SFX[ type ].checkbox.checked;
+//     // Get LocalStorage State Function
+//     SFX[ type ].getState.ls = () => window.localStorage[ SFX[ type ].localStorageTag ];
+  
+//     // Set Checkbox State Function
+//     SFX[ type ].setState.cb = ( s ) => SFX[ type ].checkbox.checked = s;
+//     // Set LocalStorage State Function
+//     SFX[ type ].setState.ls = ( s ) => window.localStorage.setItem( SFX[ type ].localStorageTag, s );
+  
+//     // Check if this has been set by user before
+//     SFX[ type ].isEnabled = SFX[ type ].getState.ls() === 'true';
+  
+//     // If not, we disable by default (and set it in localStorage)
+//     if ( SFX[ type ].getState.ls() === undefined ) {
+//       SFX[ type ].isEnabled = false;
+//       SFX[ type ].setState.ls( false );
+//     }
+  
+//     // Reflect state on the checkbox itself
+//     SFX[ type ].setState.cb( SFX[ type ].isEnabled );
+  
+//     // Attach listener
+//     $( document ).on( 'change', `#${ SFX[ type ].checkboxId }[type=checkbox]`, function () {
+//       let state = SFX[ type ].getState.cb();
+//       // Updates LocalStorage item to match checkbox state
+//        SFX[ type ].setState.ls( state );
+//        SFX[ type ].isEnabled = state;
+//     });
+// });
+
+// 2024-06-24 Miles - temporarily manually setting sfx options for testing.
+SFX.global.isEnabled = true;
+SFX.mod.isEnabled = true;
+SFX.stack.isEnabled = true;
+
+// SFX Library
+/**
+* Can build this out to be something configurable
+* in the channel settings.
+*/
+SFX.mod.sounds = {
+    eugh: {
+        src: "https://ark.augint.net/CH_Emote_SFX-eugh/CH_Emote_SFX-eugh.wav",
+        emote: "/eugh"
+    },
+    gunshot: {
+        src: "https://dontcodethis.com/images/Shotgun_Blast.wav",
+        emote: "/maths",
+    },
+    boogie: {
+        src: "https://dontcodethis.com/images/Boogie warning shot.wav",
```

Comment:
nit: This isn't a valid URL. This _probably_ works but it should be encoded, yea?